README.md

```diff
@@ -1,2 +1,145 @@
 # tally-cpp
-C++ tally client
+
+Fast, buffered, hierarchical stats collection in C++.
+
+## Abstract
+
+Tally provides a common interface for emitting metrics, while letting you not worry about the
+velocity of metrics emission.
+
+By default it buffers counters, gauges and histograms at a specified interval, but does not
+buffer timer values.  This is primarily so timer values can have all their values sampled
+if desired and if not they can be sampled as summaries or histograms independently by a reporter.
+
+## Structure
+
+- **Scope**: Keeps track of metrics and their common metadata.
+- **Metrics**: Counters, Gauges, Timers and Histograms.
+- **Reporter**: Implemented by you. Forwards aggregated values from a scope to your metrics
+ingestion pipeline. A default implementation for emitting tagged metrics using Thrift is provided.
+
+### Usage
+
+```c++
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include <m3/reporter.h>
+#include <m3/reporter_builder.h>
+#include <tally/buckets.h>
+#include <tally/scope.h>
+#include <tally/scope_builder.h>
+
+int main(int argc, char** argv) {
+  // Create a reporter that will be used to emit your metrics.
+  auto reporter =
+      m3::ReporterBuilder()
+          .host("127.0.0.1")
+          .port(9052)
+          .common_tags({{"service", "statsdex"}, {"env", "production"}})
```

Comment:
how do you want to handle default env var configuration for these values?

Reply:
I'm of two minds here. The first is that such a thing should be left to the client and the library shouldn't make any assumptions about environment variables. On the other hand, it would be easier for most users if the library did it automatically. It might be worth getting input from the ATG folks here as they will likely be the main users of the library.

Reply:
for the Java client, Justin maintains most of the code GH, all the uber specific stuff is kept in a downstream wrapper - for things like env vars, etc. I think it's fair to have just a ctor definition in the ATG codebase with that kinda thing.